src/update-done/update-done.c

```diff
@@ -50,20 +51,20 @@ static int apply_timestamp(const char *path, struct timespec *ts) {
         if (r < 0)
                 return log_error_errno(r, "Failed to set SELinux context for %s: %m", path);
 
-        fd = open(path, O_CREAT|O_WRONLY|O_TRUNC|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0644);
+        fd = open_tmpfile_linkable(path, O_CREAT|O_WRONLY|O_TRUNC|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, &tmp);
```

Comment:
O_CREAT, O_TRUNC, O_NOFOLLOW, O_NOCTTY are redundant when you call open_tmpfile_linkable(), please drop them